zerver/lib/onboarding.py

```diff
@@ -109,18 +120,49 @@ def send_initial_pms(user: UserProfile) -> None:
     )
 
 
-def bot_commands(no_help_command: bool = False) -> str:
-    commands = [
-        "apps",
-        "profile",
-        "theme",
-        "streams",
-        "topics",
-        "message formatting",
-        "keyboard shortcuts",
-    ]
-    if not no_help_command:
-        commands.append("help")
+def is_first_correct_msg(welcome_bot: UserProfile, user: UserProfile) -> bool:
+    message_queryset = Message.objects.select_for_update().filter(
+        sender=user,
+        recipient_id=welcome_bot.recipient.id,
+        content__in=WELCOME_BOT_COMMANDS,
+    )
+    if message_queryset.count() < 2:
+        return True
+    return False
+
+
+def get_confusion_emoji(welcome_bot: UserProfile, user: UserProfile) -> str:
+    # Given Welcome Bot and user, return an emoji that
+    # indicates confusion by cyling through a list of confusion-type emojis.
+    emojis = ["question", "oh_no", "grey_question"]
+    last_emoji = Reaction.objects.filter(
+        user_profile=welcome_bot, message__sender=user, emoji_name__in=emojis
+    ).last()
+    if not last_emoji:
+        return emojis[0]
+    next_emoji_index = emojis.index(last_emoji.emoji_name) + 1
+    if next_emoji_index == len(emojis):
+        return emojis[0]
+    else:
+        return emojis[next_emoji_index]
+
+
+def select_emoji_reaction(
```

Comment:
I'd name this something like `choose_welcome_bot_emoji_reaction` (since `select` sounds like it is about  `SELECT` in the SQL sense, which is not the point of the name after all) to be a bit clearer about what this is

Reply:
fixed.